Resources/ArmTemplates/template.json

```diff
@@ -0,0 +1,350 @@
+          "servicetaxonomy-apim-logger"
+        ],
+        "properties": {
+            "mode": "Incremental",
+            "templateLink": {
+                "uri": "[concat(variables('BuildingBlocksDfcBaseUrl'), 'APIM/apim-product.json')]",
+                "contentVersion": "1.0.0.0"
+            },
+            "parameters": {
+                "allowAnonymousAccess": {
+                    "value": false
+                },
+                "apimServiceName": {
+                    "value": "[parameters('ApimServiceName')]"
+                },
+                "productDisplayName": {
+                    "value": "[variables('ApimProductName')]"
+                }
+            }
+        }
+    },
+    {
+      "apiVersion": "2017-05-10",
+      "name": "[concat('service-taxononmy-', variables('allFunctions')[copyIndex()].functionName)]",
+      "resourceGroup": "[parameters('ApimResourceGroup')]",
+      "type": "Microsoft.Resources/deployments",
+      "dependsOn": [
+        "service-taxonomy-apim-product",
+        "servicetaxonomy-apim-logger"
+      ],
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[concat(variables('BuildingBlocksDfcBaseUrl'),'APIM/apim-api.json')]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "apimServiceName": {
+            "value": "[parameters('ApimServiceName')]"
+          },
+          "apiName": {
+            "value": "[variables('allFunctions')[copyIndex()].functionType]"
+          },
+          "apimProductInstanceName": {
+            "value": "service-taxonomy"
+          },
+          "apimLoggerName": {
+            "value": "[reference('servicetaxonomy-apim-logger').outputs.ApimLoggerName.value]"
+          }
+        }
+      },
+      "copy": {
+        "name": "apiCount",
+        "count": "[length(variables('allFunctions'))]"
+      }
+    },
+    {
+      "apiVersion": "2017-05-10",
+      "name": "[variables('allFunctions')[copyIndex()].appInsightsName]",
+      "type": "Microsoft.Resources/deployments",
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[concat(variables('BuildingBlocksDfcBaseUrl'),'application-insights.json')]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "appInsightsName": {
+            "value": "[variables('allFunctions')[copyIndex()].appInsightsName]"
+          },
+          "attachedService": {
+            "value": "[variables('allFunctions')[copyIndex()].functionName]"
+          }
+        }
+      },
+      "copy": {
+        "name": "appInsightsCount",
+        "count": "[length(variables('allFunctions'))]"
+      }
+    },
+    {
+      "apiVersion": "2017-05-10",
+      "name": "[variables('allFunctions')[copyIndex()].functionName]",
+      "type": "Microsoft.Resources/deployments",
+      "dependsOn": [
+        "[variables('allFunctions')[copyIndex()].appInsightsName]",
```

Comment:
You can use the copy name (ie "appInsightsCount") as a dependency, that might be a clearer way of doing this.  DYSAC uses that method successfully

Reply:
Updated to use the copy name